Bug: [flat config] Global `ignores` matches differently from non-global `ignores`

### Environment

Node version: v18.12.1
npm version: 8.19.2
Local ESLint version: v8.41.0
Global ESLint version: none
Operating System: Linux or MacOS

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

##### `eslint-global-ignore.config.js`

```js
export default [
  {
    ignores: ['ignored/'],
  },
  {
    rules: {
      'no-undef': 'error',
    },
  },
];
```

##### `eslint-files-ignore.config.js`

```js
export default [
  {
    files: ['**/*.js'],
    ignores: ['ignored/'],
    rules: {
      'no-undef': 'error',
    },
  },
];
```
</details>

##### `file.js` and `ignored/file.js`

```js
console.log();
```

### What did you expect to happen?

Not sure what is the correct intended behavior here, so raising this issue for the maintainers' consideration. Based on the documentation ([Globally ignoring files with `ignores`](https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores)), it looks like `ignores` can match and ignore an entire directory tree:

```js
{
  // Ignores directory tree
  ignores: ["ignored/"],
}
```

However, if the same `ignores` pattern is used in a configuration block which also has `files`, then the pattern is interpreted completely differently, and does not have the same effect:

```js
{
  files: ["**/*.js"],
  // Does *not* ignore directory tree
  ignores: ["ignored/"],
}
```

Rather, to ignore the entire top-level `ignored` tree, the `ignores` pattern needs to be adjusted as follows:

```diff
 {
   files: ["**/*.js"],
-  ignores: ["ignored/"],
+  ignores: ["ignored/**"], 
 }
```

Is this difference in behavior between the two `ignores` (global and non-global) intentional? And if so, can it be explicitly documented, if not already? It seems like a surprising "gotcha".

### What actually happened?



```
$ ESLINT_USE_FLAT_CONFIG=true eslint -c eslint-global-ignore.config.js .

/repo/file.js
  1:1   error  'console' is not defined  no-undef

✖ 1 problems (1 errors, 0 warnings)
```

```
$ ESLINT_USE_FLAT_CONFIG=true eslint -c eslint-files-ignore.config.js .

/repo/ignored/file.js
  1:1   error  'console' is not defined  no-undef

/repo/file.js
  1:1   error  'console' is not defined  no-undef

✖ 2 problems (2 errors, 0 warnings)
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/eslint-ignore-pattern-inconsistency

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

_No response_

--- a/docs/src/use/configure/configuration-files-new.md
+++ b/docs/src/use/configure/configuration-files-new.md
@@ -116,6 +116,8 @@ export default [
 
 Here, the configuration object excludes files ending with `.config.js` except for `eslint.config.js`. That file still has `semi` applied.
 
+Non-global `ignores` patterns can only match file names. A pattern like `"dir-to-exclude/"` will not ignore anything. To ignore everything in a particular directory, a pattern like `"dir-to-exclude/**"` should be used instead.
+
 If `ignores` is used without `files` and there are other keys (such as `rules`), then the configuration object applies to all files except the ones specified in `ignores`, for example:
 
 ```js

--- a/docs/src/use/configure/configuration-files-new.md
+++ b/docs/src/use/configure/configuration-files-new.md
@@ -116,7 +116,7 @@ export default [
 
 Here, the configuration object excludes files ending with `.config.js` except for `eslint.config.js`. That file still has `semi` applied.
 
-If `ignores` is used without `files` and any other setting, then the configuration object applies to all files except the ones specified in `ignores`, for example:
+If `ignores` is used without `files` and there are other keys (such as `rules`), then the configuration object applies to all files except the ones specified in `ignores`, for example:
 
 ```js
 export default [
@@ -144,6 +144,9 @@ export default [
 ```
 
 This configuration specifies that all of the files in the `.config` directory should be ignored. This pattern is added after the default patterns, which are `["**/node_modules/**", ".git/**"]`.
+
+Note that only global `ignores` patterns can ignore or unignore whole directories.
+`ignores` patterns that are specific to a configuration will only match file names.
 
 #### Cascading configuration objects
 

--- a/docs/src/use/configure/configuration-files-new.md
+++ b/docs/src/use/configure/configuration-files-new.md
@@ -147,7 +147,7 @@ export default [
 
 This configuration specifies that all of the files in the `.config` directory should be ignored. This pattern is added after the default patterns, which are `["**/node_modules/**", ".git/**"]`.
 
-Note that only global `ignores` patterns can ignore or unignore whole directories.
+Note that only global `ignores` patterns can match directories.
 `ignores` patterns that are specific to a configuration will only match file names.
 
 #### Cascading configuration objects

